modules/openxr/register_types.cpp

```diff
@@ -225,10 +225,16 @@ void initialize_openxr_module(ModuleInitializationLevel p_level) {
 		}
 
 #ifdef TOOLS_ENABLED
+		// Register as "editor", not "core".
+		ClassDB::APIType prev_api = ClassDB::get_current_api();
+		ClassDB::set_current_api(ClassDB::API_EDITOR);
+
 		GDREGISTER_ABSTRACT_CLASS(OpenXRInteractionProfileEditorBase);
 		GDREGISTER_CLASS(OpenXRInteractionProfileEditor);
 		GDREGISTER_CLASS(OpenXRBindingModifierEditor);
 
+		ClassDB::set_current_api(prev_api);
+
```

Comment:
Why mess with `ClassDB::set_current_api()` rather than putting this stuff in a new `if (p_level == MODULE_INITIALIZATION_LEVEL_EDITOR) { ... }` block?

Reply:
As mentioned, I took inspiration from https://github.com/godotengine/godot/pull/86209/files.

Should that one be changed, too?

Reply:
If there isn't a technical reason to do it that way, then, yeah, that one should be changed too, since that's not how the API is intended to be used. (We don't need to do that on this PR, though.)

Reply:
We seem to have a bunch of cases that need to be cleaned up in a follow up PR:
```
$ rg -g'register_types.cpp' -l API_EDITOR 
fbx/register_types.cpp
gltf/register_types.cpp
minimp3/register_types.cpp
gdscript/register_types.cpp
vorbis/register_types.cpp
```

And many of those (and maybe others) don't seem to use the `p_level` consistently. Doesn't this actually lead to registering classes multiple times when this is called for each initialization level?

Reply:
I'll merge this now to include in my tentative build for 4.4.1-rc1, but we can improve registration of modules in a follow-up.

Reply:
I just made PR https://github.com/godotengine/godot/pull/104084 to register these editor classes in the correct way. (We can keep that in 4.5 if we want - the code in 4.4 works fine, even if it's using the API incorrectly.)

> And many of those (and maybe others) don't seem to use the `p_level` consistently. Doesn't this actually lead to registering classes multiple times when this is called for each initialization level?

It looks like those only register their classes once. Some just have this at the start of their register function:

```
	if (p_level != MODULE_INITIALIZATION_LEVEL_SCENE) {
		return;
	}
```